File: src/main/java/apiservice/model/History.java
package apiservice.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "history")
public class History {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name="bench_id", nullable=false)
    private Bench bench_;

    @ManyToOne
    @JoinColumn(name="location_id", nullable=false)
    private Location location_;

    @ManyToOne
    @JoinColumn(name="status_id", nullable=false)
    private Status status_;
}

File: backend/DAL/Models/History.cs
using Microsoft.CodeAnalysis;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace backend.DAL.Models
{
    public class History
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public int BenchId { get; set; }
        public Bench Bench { get; set; }

        public int LocationId { get; set; }
        public Location Location { get; set; }

        public int StatusId { get; set; }
        public Status Status { get; set; }


    }
}
